relay-redis/src/scripts/is_rate_limited.lua

```diff
@@ -71,9 +71,14 @@ if not failed then
         local k = i * 2 + 1
         local v = i * 4 + 1
 
-        if tonumber(ARGV[v + 2]) > 0 then
-            redis.call('INCRBY', KEYS[k], ARGV[v + 2])
-            redis.call('EXPIREAT', KEYS[k], ARGV[v + 1])
+        local quantity = tonumber(ARGV[v + 2])
+        local expiry = ARGV[v + 1]
+
+        if quantity > 0 then
+            if redis.call('INCRBY', KEYS[k], quantity) == quantity then
```

Comment:
This is intended to test that the previous value was 0, so this is the first increment, correct?

Reply:
Correct, incrby returns the new value, the new value only matches the quantity if there was a 0 before.